### PR TITLE
Update README, in preparation for GA release

### PR DESCRIPTION
**Pull request by jasonblais**
Added
 - an intro with feature description and screenshots
 - detailed descriptions of the various parameters used in config.json
 - a more generic example for new team member onboarding (would be good to add screenshots for this example later)
 - standard development section

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,126 +1,158 @@
...
+*Welcome a new team member to Mattermost Contributors team. Then add the user to a set of channels based on their selection.*
+
+![image](https://user-images.githubusercontent.com/13119842/58736540-30fd8980-83cc-11e9-8e8e-94ea3042b3b1.png)
+
+## Configuration
+
+1. Go to **System Console > Plugins > Management** and click **Enable** to enable the Welcome Bot plugin.
+    - If you are running Mattermost v5.11 or earlier, you must first go to the [releases page of this GitHub repository](https://github.com/mattermost/mattermost-plugin-welcomebot/releases), download the latest release, and upload it to your Mattermost instance [following this documentation](https://docs.mattermost.com/administration/plugins.html#plugin-uploads).
+
+2. Modify your `config.json` file to include your custom attributes, under the `PluginSettings`. See below for an example of what this should look like.
```
Comment (cpoile) [inline README.md]: 1/5 `your custom attributes` is confusing. Maybe `your Welcome Bot's messages and actions`?

File: README.md [hunk 2]
```diff
@@ -1,126 +1,158 @@
...
+        "AttachmentMessage": [
+            "Attachment message containing user actions"
+        ],
+        "Actions" : [
+            {
+                "ActionType": "button",
+                "ActionDisplayName": "User Action",
+                "ActionName": "action-name",
+                "ActionSuccessfulMessage": [
+                    "Message posted after use takes this action and joins channels specified by 'ChannelsAddedTo'."
```
Comment (cpoile) [inline README.md]:
```suggestion
                    "Message posted after the user takes this action and joins channels specified by 'ChannelsAddedTo'."
```

File: README.md [hunk 3]
```diff
@@ -1,126 +1,158 @@
...
+                "ChannelsAddedTo": ["channel3", "channel4"]
+            }
+        ]
+    }
+]
+```
+
+where
+
+- **TeamName**: The team for which the Welcome Bot sends a message for. Refers to the team handle used in the URL. For example, in the following URLm the **TeamName** value is `myteam`: https://example.com/myteam/channels/mychannel
```
Comment (cpoile) [inline README.md]: 👍 nice detailed examples

File: README.md [hunk 4]
```diff
@@ -1,126 +1,158 @@
...
+- **DelayInSeconds**: The number of seconds after joining a team that the user receives a welcome message.
+- **Message**: The message posted to the user.
+- (Optional) **AttachmentMessage**: Message text in attachment containing user action buttons. 
+- (Optional) **Actions**: Use this to add new team members to channels automatically or based on which action button they pressed.
+    - **ActionType**: One of `button` or `automatic`. When a `button`, enables uses to select which types of channels they want to join. When `automatic`, the user is automatically added to the specified channels.
```
Comment (cpoile) [inline README.md]:
```suggestion
   - **ActionType**: One of `button` or `automatic`. When `button`: enables uses to select which types of channels they want to join. When `automatic`: the user is automatically added to the specified channels.
```

File: README.md [hunk 5]
```diff
@@ -1,126 +1,158 @@
...
+    - **ActionDisplayName**: Sets the display name for the user action buttons.
+    - **ActionName**: Sets the action name used by the plugin to identify which action is taken by a user.
+    - **ActionSuccessfulMessage**: Message posted after use takes this action and joins the specified channels.
```
Comment (cpoile) [inline README.md]:
```suggestion
    - **ActionSuccessfulMessage**: Message posted after the user takes this action and joins the specified channels.
```

File: README.md [hunk 6]
```diff
@@ -1,126 +1,158 @@
...
                         "ActionSuccessfulMessage": [
-                            "### Baller, I have added you to the following developer related channels!",
-                            "~developers - Great for general developer questions",
-                            "~developer-toolkit - Great questions about plugins or integrations",
-                            "~developer-meeting - Weekly core staff and community meeting",
-                            "~developer-performance - Great for questions about performance or load testing",
-                            "~bugs - To help investigate or report bugs"
+                            "### Awesome! I have added you to the following developer channels:",
+                            "~leads - To stay updated on incoming leads",
+                            "~help-desk - To connect with your support team for quick Q&A
```
Comment (cpoile) [inline README.md]:
```suggestion
```

File: README.md [hunk 7]
```diff
@@ -1,126 +1,158 @@
...
                         "ActionSuccessfulMessage": [
-                            "### Great, I have added you to a few channels that might be interesting!",
-                            "~peer-to-peer-help - General help and setup",
-                            "~feature-ideas - To discuss potential feature ideas",
-                            "~developers - Great for general developer questions",
-                            "~bugs - To help investigate or report bugs"
+                            "### Awesome! I have added you to the following developer channels:",
+                            "~bugs - To help investigate or report bugs",
+                            "~customer-support - To troubleshoot and resolve customer issues",
+                            "~help-desk - To connect with account managers for quick Q&A"
```
Comment (cpoile) [inline README.md]:
```suggestion
                            "~leads - To discuss potential accounts with other account managers"
```

**Comment by cpoile** [comment]
We also should have a description of the template values you can insert into the messages. 
Something like:

We've used `{{.UserDisplayName}}` and `{{.Team.DisplayName}}' in the example `config.json`. You can insert any variable from the `MessageTemplate` struct, which has the following fields:
```go
type MessageTemplate struct {
	WelcomeBot      *model.User
	User            *model.User
	Team            *model.Team
	Townsquare      *model.Channel
	DirectMessage   *model.Channel
	UserDisplayName string
}
```

File: README.md (hunk 1 above)
Comment (jasonblais) [reply README.md]: Good catch, copy-paste error :)

**Comment by jasonblais** [comment]
Great feedback @cpoile! Updated

**Pull request merged**